reactive/kubernetes_master.py

```diff
@@ -570,11 +578,11 @@ def setup_leader_authentication():
     # Try first to fetch data from an old leadership broadcast.
     if not get_keys_from_leader(keys) \
             or is_state('reconfigure.authentication.setup'):
-        if not os.path.isfile(basic_auth):
-            kubernetes_master.deprecate_auth_file(basic_auth)
+        kubernetes_master.deprecate_auth_file(basic_auth)
+        set_flag('kubernetes-master.basic-auth.migrated')
```

Comment:
It confused me at first that the upgrade handler is checking the return value of the deprecate function but this is not.  But I don't think it actually matters in practice.

Reply:
Technically, the upgrade handler is checking the return of *migrate_auth_file*, not *deprecate_auth_file*.  The thinking here is that you'll only get to this code if:

1. Brand new deployment with no leader keys; in this case, we always want to write the deprecated file and, more importantly, set the `.migrated` flag so we know to check for secrets from the get-go.
2. Upgraded deployment at which point `upgrade-charm` will have already done the migration, so checking `deprecate` status seems moo, like a cow's opinion.

Reply:
Right you are.  My mistake.